frontend/src/components/CustomerDetails.cy.js

```diff
@@ -0,0 +1,92 @@
+/* global cy */
+
+import * as React from 'react'
+import CustomerDetails from './CustomerDetails'
+import '../index.css'
+
+const customer = {
+  name: 'JonQA',
+  employees: 500,
+  size: 'large',
+  contactInfo: {
+    name: 'Jon Vendas',
+    email: 'john.qa@gmail.com'
+  },
+  address: {
+    street: '123 Main St',
+    city: 'Metropolis',
+    state: 'NY',
+    zipCode: '10001',
+    country: 'USA'
+  }
+}
+const customerWithoutContactInfo = {
+  name: 'JonSDT',
+  employees: 1,
+  size: 'smal',
+  contactInfo: null,
+  address: {
+    street: '123 Main St',
+    city: 'Metropolis',
+    state: 'NY',
+    zipCode: '10001',
+    country: 'USA'
+  }
+}
+
+const customerWithoutAddress = {
+  name: 'NoAddressCustomer',
+  employees: 10,
+  size: 'small',
+  contactInfo: {
+    name: 'No Address',
+    email: 'no.address@example.com'
+  },
+  address: null
+}
+
+describe('CustomerDetails component', () => {
+  it('Shows the contact info of a specific customer', () => {
+    cy.mount(<CustomerDetails customer={customer} />)
+    cy.get('.customer-details p').contains(`Contact name: ${customer.contactInfo.name}`).should('be.visible')
```

Comment:
Que tal algo assim?

```suggestion
    cy.contains('p', `Contact name: ${customer.contactInfo.name}`).should('be.visible')
```

Essa format é mais robusta, visto como o Cypress trabalha com retentativas.

> O mesmo vale para as próximas verificações.